Geometry/HGCalCommonData/src/HGCalParametersFromDD.cc

```diff
@@ -52,11 +59,34 @@ bool HGCalParametersFromDD::build(const DDCompactView* cpv,
   DDSpecificsMatchesValueFilter filter{val};
   DDFilteredView fv(*cpv,filter);
   bool ok = fv.firstChild();
+  int  mode(0);
 
   if (ok) {
     DDsvalues_type sv(fv.mergedSpecifics());
-    php.mode_ = getGeometryMode("GeometryMode", sv);
+    php.mode_ = getGeometryMode("GeometryMode", sv, true);
```

Comment:
ok - seems I understand better now - clearly the better solution is to avoid the extract bool in getGeometryMode and just have two functions, one for each type of enum returned by getGeometryMode. This avoids the if statement in getGeometryMode (which is much of the function, so no big code reuse benefit anyway), and avoids lots of casting.